code/data_structures/src/binary_heap/binary_heap.cpp

```diff
@@ -0,0 +1,175 @@
+/* Binary Heap in C++ */
+
+#include <iostream>
+#include <cstdlib>
+#include <vector>
+#include <iterator>
+
+
+class BinaryHeap{
+
+    private:
+        std::vector <int> heap;
+        int left(int parent);
+        int right(int parent);
+        int parent(int child);
+        void heapifyup(int index);
+        void heapifydown(int index);
+    public:
+        BinaryHeap()
+        {}
+        void Insert(int element);
+        void DeleteMin();
+        int ExtractMin();
+        void DisplayHeap();
+        int Size();
+};
+
+// Return Heap Size
+int BinaryHeap::Size(){
+    return heap.size();
+}
+ 
+// Insert Element into a Heap 
+void BinaryHeap::Insert(int element){
+    heap.push_back(element);
+    heapifyup(heap.size() -1);
+}
+
+// Delete Minimum Element 
+void BinaryHeap::DeleteMin(){
+    
+    if (heap.size() == 0){
+        std::cout << "Heap is Empty"<< std::endl;
+        return;
+    }
+    heap[0] = heap.at(heap.size() - 1);
+    heap.pop_back();
+    heapifydown(0);
+    std::cout << "Element Deleted" << std::endl;
+}
+ 
+// Extract Minimum Element
+int BinaryHeap::ExtractMin(){
+    if (heap.size() == 0){
+        return -1;
+    }
+    else
+        return heap.front();
+}
+ 
+// Display Heap
+void BinaryHeap::DisplayHeap(){
+    std::vector <int>::iterator pos = heap.begin();
+
+    std::cout << "Heap -->  ";
+    while (pos != heap.end()){
```

Comment:
Just use a range-for:

```
for (const auto& elem : heap)
    os << elem << " ";
```

`os` should be the ostream iterator provided by `<<`.